cjsmin/src/chess.ts

```diff
@@ -1649,6 +1671,55 @@ export class Chess {
     }
   }
 
+  *historyGenerator(
+    pgnMoveLine: string,
+    { strict = false }: { strict?: boolean; newlineChar?: string } = {}
+  ): Generator<{ move: PrettyMove; board: Array<Piece> }, void, void> {
+    this.load(DEFAULT_POSITION);
+
+    // We don't mind destructive deletion of the comments
+    let ms = pgnMoveLine.replace(new RegExp(`({[^}]*})+?`, 'g'), '');
+
+    // delete move numbers
+    ms = ms.replace(/\d+\.(\.\.)?/g, '');
+
+    // delete ... indicating black to move
+    ms = ms.replace('...', '');
+
+    /* delete numeric annotation glyphs */
+    ms = ms.replace(/\$\d+/g, '');
+
+    // trim and get array of moves
+    // let moves = ms.trim().split(new RegExp(/\s+/));
+    let moves = ms.trim().split(' ');
+
+    // delete empty entries
+    moves = moves.filter((move) => move !== '');
+
+    for (let halfMove = 0; halfMove < moves.length; halfMove++) {
+      const move = this._moveFromSan(moves[halfMove], strict);
+
+      // invalid move
+      if (move == null) {
+        // was the move an end of game marker
+        if (!(TERMINATION_MARKERS.indexOf(moves[halfMove]) > -1)) {
+          throw new Error(`Invalid move in PGN: ${moves[halfMove]}`);
+        }
+      } else {
+        // reset the end of game marker if making a valid move
+        this._makeMove(move, moves[halfMove]);
+        const prettyMove = {
+          ...this._makePretty(move, moves[halfMove]),
+          originalString: moves[halfMove],
+        };
+        yield {
+          move: prettyMove,
+          board: this._board,
+        };
+      }
+    }
+  }
+
```

Comment:
this code takes a PGN string, processes it move by move, converts each move to an internal representation, updates the chessboard, and yields the resulting board state along with information about the move